### PR TITLE
#4296 name note for all vaccinations

### LINKED ISSUE #4296: Create a vaccination name note when creating a vaccination prescription
## Is your feature request related to a problem? Please describe.

- Want to track vaccinations of patients without having their transactions

## Describe the solution you'd like

- When creating a prescription, create a name_note with the patient event ' 🤷  ' which has the vaccination event details

## Implementation

Unsureee!

## Describe alternatives you've considered

- Add an asterix to the patients name

## Additional context

- https://github.com/sussol/msupply/issues/9253

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #4296 

Note: I haven't tested sync yet, sorry- will do soon. I am in the middle of creating 9million sync out records for testing and it's taking awhile.

## Change summary

- Just changes from creating a refusal name note when refused to creating a vaccination name note, everytime.

## Testing

- [ ] Create a vaccine prescription where everything is standard. Sync and view the name note on desktop (Through record browser :))
- [ ] Create a vaccine prescription where you refuse the vaccine. Sync and view the name note on desktop (Through record browser :))
- [ ] Create a vaccine prescription where you use a bonus dose. Sync and view the name note on desktop (Through record browser :))

### Related areas to think about

- Too much info? :)

### CODE REVIEW

File: src/actions/Entities/VaccinePrescriptionActions.js [hunk 1]
```diff
@@ -163,13 +163,41 @@ const createPrescription = (
...
+    vaccine: selectedBatch?.itemName,
+    vaccineCode: selectedBatch?.itemCode,
+    batch: selectedBatch?.batch,
+    expiry: selectedBatch?.expiryDate,
+    vaccinator: vaccinator?.displayString,
+    extra: {
+      prescription: prescription?.toJSON(),
+      vaccinator: vaccinator.toJSON(),
+      patient: patient.toJSON(),
+    },
```
Comment (joshxg) [inline src/actions/Entities/VaccinePrescriptionActions.js:191]: 🤷

Comment (mark-prins) [reply src/actions/Entities/VaccinePrescriptionActions.js:191]: 😄  I tested first - and saw all this on the server! 
sure, why not..

File: src/actions/Entities/VaccinePrescriptionActions.js [hunk 2]
```diff
@@ -163,13 +163,41 @@ const createPrescription = (
...
-  const [patientEvent] = UIDatabase.objects('PatientEvent').filtered('code == "RV"');
+const createVaccinationNameNote = (
+  patient,
+  prescription,
+  refused,
+  bonusDose,
+  vaccinator,
+  selectedBatch
+) => {
+  const [patientEvent] = UIDatabase.objects('PatientEvent').filtered('code == "vaccination"');
```
Comment (mark-prins) [inline src/actions/Entities/VaccinePrescriptionActions.js:174]: I tested first without creating this patient event 🙄

**Review by mark-prins — APPROVED** [review]
😄  A thing of beauty!

**Comment by mark-prins** [comment]
@joshxg I've remembered [this one](https://github.com/openmsupply/mobile/pull/4242) which adds the refusal reason - also requested by Kiribati. I see now why I missed it - this PR is for _develop_ and the other is _8.2.0_
We'll need to add a refusal reason element when merging develop back in to 8.2 prior to release.

**Comment by joshxg** [comment]
Ah- thanks- i'll make a change for that now

**Pull request merged**